modules/mapping-utils/src/resolveSets.js

```diff
@@ -149,48 +175,193 @@ export const deleteSaveSets = ({ callback }) => async (
   return esResponse.deleted;
 };
 
-export const renameSaveSetTag = ({ callback }) => async (
+export const updateSet = ({ types, callback }) => async (
   obj,
-  { setId, tag, userId },
+  { source, subAction, target, userId, data },
   { es },
 ) => {
-  const esResponse = await es.updateByQuery({
-    index: CONSTANTS.ES_ARRANGER_SET_INDEX,
-    type: CONSTANTS.ES_ARRANGER_SET_TYPE,
-    refresh: true,
-    body: {
-      script: {
-        lang: 'painless',
-        source: `ctx._source['tag'] = '${tag}'`,
-      },
-      query: {
-        bool: {
-          filter: {
-            term: { userId: userId },
+  const { sourceType } = source;
+
+  switch (subAction) {
+    case SubActionTypes.REMOVE_IDS:
+    case SubActionTypes.ADD_IDS: {
+      if (SourceType.QUERY === sourceType) {
+        const { type, sqon, path } = data;
+
+        if (isQueryEmpty(sqon)) {
+          throw new Error('Query must not be empty');
+        }
+
+        const { setId } = target;
+
+        const esSearchResponse = await es.search({
+          index: CONSTANTS.ES_ARRANGER_SET_INDEX,
+          type: CONSTANTS.ES_ARRANGER_SET_TYPE,
+          body: {
+            query: {
+              bool: {
+                filter: {
+                  term: { userId: userId },
+                },
+                must: {
+                  term: {
+                    setId: {
+                      value: setId,
+                    },
+                  },
+                },
+              },
+            },
           },
-          must: {
-            term: {
-              setId: {
-                value: setId,
+        });
+
+        const sets = mapHits(esSearchResponse);
+        if (sets.length === 0) {
+          throw new Error('Internal error');
```

Comment:
why having no results is and exception? does it mean the system crashed? Somebody cut a wired?

Reply:
haha good point